### ISSUE TITLE
typeUnparen false positive for `<- chan`

### ISSUE THREAD

**Issue by ian-h-chamberlain** [go-critic/go-critic#1428]
Looks to be similar cause as #1035 but a slightly different case.
Tested on `v0.11.5-0.20240722075729-97e2bc38fa41`.

`typeUnparen` suggests removing parentheses around `<-chan` types that would actually break the code, e.g.
```go
package main

func main() {
	a := (<-chan int)(nil)
	recv(a)
	b := (chan<- int)(nil)
	send(b)
}

func recv(c <-chan int) {
	<-c
}

func send(c chan<- int) {
	c <- 0
}
```
```console
$ gocritic check -enableAll
./main.go:4:7: typeUnparen: could simplify (<-chan int) to <-chan int
./main.go:6:7: typeUnparen: could simplify (chan<- int) to chan<- int
$ 
```

Following the suggest changes, we get a compiler error because `a := <-chan int(nil)` is now assigning the result of `<-nil`, an int value, to `a`.

```console
$ go build main.go
# command-line-arguments
./main.go:5:7: cannot use a (variable of type int) as <-chan int value in argument to recv
```

I guess `chan<- int` is less ambiguous so isn't broken the same way, although it's arguable that keeping parentheses there would be more readable...
